src/utils/translate.py

```diff
@@ -92,28 +93,34 @@ def format_inputs(
     timesteps = torch.arange(start=window_start, end=window_start + window_size, device=device).reshape(1, window_size)
     attention_mask = torch.ones(1, window_size, device=device, dtype=torch.float32)  # Needed for padding
```

Comment:
Are you sure that this attention mask accomplishes anything? After looking through the modeling_decision_transformer.py code, it seems to me that 1 is subtracted from the attention mask, making an array of zeros, which are then added to the attention weights. This effectively changes nothing. I'm not sure of the original intention of this attention mask, but I think we also have a problem where there is no autoregressive mask applied to the attention weights. The attention heads can then attend to states further on in the sequence.

Reply:
It is the default setting, so not compulsory (I can remove it), but it could be useful for other experiments to not attend the whole sequence.

Reply:
For the rest, I don't see how you jumped to this conclusion. We can discuss this on Discord.